jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/ProjectProperties.java

```diff
@@ -48,29 +51,36 @@ String getMainClass(@Nullable String mainClass) {
     if (mainClass == null) {
       mainClass = getMainClassFromJarTask();
       if (mainClass == null) {
-        gradleBuildLogger.info(
+        gradleBuildLogger.debug(
             "Could not find main class specified in a 'jar' task; attempting to "
                 + "infer main class.");
+
+        final String mainClassSuggestion = "add a `mainClass` configuration to jib";
         try {
-          mainClass =
-              MainClassFinder.findMainClass(
-                  project
-                      .getBuildDir()
-                      .toPath()
-                      .resolve("classes")
-                      .resolve("java")
-                      .resolve("main")
-                      .toAbsolutePath()
-                      .toString());
-        } catch (MultipleClassesFoundException | IOException ex) {
-          throw new GradleException(
-              HelpfulSuggestionsProvider.get("Failed to get main class: " + ex.getMessage())
-                  .suggest("add a `mainClass` configuration to jib"));
-        }
-        if (mainClass == null) {
+          // Adds each file in each classes output directory to the classes files list.
+          JavaPluginConvention javaPluginConvention =
```

Comment:
It was either this or something like `getSourceFilesConfiguration().getClassesFiles().get(0).getParent()` since I need the root directory, and `getClassesFiles` seems to return a list of the files in the root directory. Plus using `getSourceFilesConfiguration()` was weird in testing, since I would either need to refactor a bit, or mock a bunch of the `GradleSourceFilesConfiguration` internals to avoid null pointer exceptions.